lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -33,20 +70,183 @@ def onCreate(self, *args):
+
+        # Update all metadata values.
+        self.refresh()
+
+    def refresh(self):
+        # PrimPath
+        cmds.textFieldGrp(self.primPath, edit=True, text=str(self.prim.GetPath()))
+
+        # Kind
+        model = Usd.ModelAPI(self.prim)
+        primKind = model.GetKind()
+        if not primKind:
+            # Special case to handle the empty string (for meta data value empty).
+            cmds.optionMenuGrp(self.kind, edit=True, select=1)
+        else:
+            cmds.optionMenuGrp(self.kind, edit=True, value=primKind)
+
+        # Active
+        cmds.checkBoxGrp(self.active, edit=True, value1=self.prim.IsActive())
+
+        # Instanceable
+        cmds.checkBoxGrp(self.instan, edit=True, value1=self.prim.IsInstanceable())
+
+        # All other metadata types
+        for k in self.extraMetadata:
+            v = self.prim.GetMetadata(k) if k != 'customData' else self.prim.GetCustomData()
+            cmds.textFieldGrp(self.extraMetadata[k], edit=True, text=str(v))
+
+    def _onKindChanged(self, value):
+        with mayaUsdLib.UsdUndoBlock():
+            model = Usd.ModelAPI(self.prim)
+            model.SetKind(value)
+
+    def _onActiveChanged(self, value):
+        with mayaUsdLib.UsdUndoBlock():
+            self.prim.SetActive(value)
+
+    def _onInstanceableChanged(self, value):
+        with mayaUsdLib.UsdUndoBlock():
+            self.prim.SetInstanceable(value)
+
+class NoticeListener(object):
+    # Inserted as a custom control, but does not have any UI. Instead we use
+    # this control to be notified from USD when any metadata has changed
+    # so we can update the AE fields.
+    def __init__(self, prim, metadataControls):
+        self.prim = prim
+        self.metadataControls = metadataControls
+
+    def onCreate(self, *args):
+        self.listener = Tf.Notice.Register(Usd.Notice.ObjectsChanged,
+                                           self.__OnPrimsChanged, self.prim.GetStage())
+        pname = cmds.setParent(q=True)
+        cmds.scriptJob(uiDeleted=[pname, self.onClose], runOnce=True)
+
+    def onReplace(self, *args):
+        pass
+
+    def onClose(self):
+        if self.listener:
+            self.listener.Revoke()
+            self.listener = None
+
+    def __OnPrimsChanged(self, notice, sender):
+        if notice.HasChangedFields(self.prim):
+            # Iterate thru all the metadata controls (we were given when created) and
+            # call the refresh method (if it exists).
+            for ctrl in self.metadataControls:
+                if hasattr(ctrl, 'refresh'):
+                    ctrl.refresh()
+
 
 # SchemaBase template class for categorization of the attributes.
-class AETemplate(ufeAeTemplate.Template):
+# We no longer use the base class ufeAeTemplate.Template as we want to control
+# the placement of the metadata at the bottom (after extra attributes).
+class AETemplate(object):
```

Comment:
This is because Nat wanted the **metadata** after the **extra attributes**. It was the base class that controlled where to put that section (from the __init__). The case class it nothing special - just a couple of helpers that I copied here.